Source/ARTHttp.m

```diff
@@ -175,18 +179,19 @@ - (void)executeRequest:(NSMutableURLRequest *)request completion:(void (^)(NSHTT
     [self.logger debug:@"%@ %@", request.HTTPMethod, request.URL.absoluteString];
     [self.logger verbose:@"Headers %@", request.allHTTPHeaderFields];
 
+    __weak typeof(self) weakSelf = self;
```

Comment:
@ricardopereira But this way it's easy that you e. g. create a ARTRest instance, fire some request expecting a result (history, etc.), then release your instance since you don't need it anymore, but you never get your results. This way you're forcing users to keep instances around not as long as _they_ are using them but as long as requests using it may take, etc. which pretty much amounts to manually managed memory. This behavior goes against what all other libraries do and what you'd expect from a garbage-collected language.